spring-pulsar/src/main/java/org/springframework/pulsar/core/DefaultSchemaResolver.java

```diff
@@ -129,8 +132,22 @@ public <T> Schema<T> getSchema(Class<?> messageClass, boolean returnDefault) {
 	}
 
 	@Nullable
-	private Schema<?> getCustomSchemaOrMaybeDefault(Class<?> messageClass, boolean returnDefault) {
-		return this.customSchemaMappings.getOrDefault(messageClass, (returnDefault ? Schema.BYTES : null));
+	protected Schema<?> getCustomSchemaOrMaybeDefault(Class<?> messageClass, boolean returnDefault) {
+		Schema<?> schema = this.customSchemaMappings.get(messageClass);
+		if (schema == null && returnDefault) {
+			if (messageClass != null) {
+				try {
+					return Schema.JSON(messageClass);
+				}
+				catch (Exception e) {
+					if (this.logger.isDebugEnabled()) {
```

Comment:
This is the beauty of `LogAccessor`.... you can ditch the "if level enabled check" w/ 
```java
this.logger.debug(() -> "Failed to create JSON schema for %s".formatted(messageClass.getName());
```

Reply:
done